src/PlaywrightSharp.Firefox/FirefoxPage.cs

```diff
@@ -246,7 +250,13 @@ public Task SetCacheEnabledAsync(bool enabled)
 
         public Task SetRequestInterceptionAsync(bool enabled) => throw new NotImplementedException();
 
-        public Task AuthenticateAsync(Credentials credentials) => throw new NotImplementedException();
+        public Task AuthenticateAsync(Credentials credentials)
+            => _session.SendAsync(
+                new NetworkSetAuthCredentialsRequest
+                {
+                    Username = credentials?.Username,

```

Comment:
we need to allow to serialize as `{ username: null, password: null }`

Reply:
Can you explain this a little bit more?

Reply:
we need to send a json that includes both properties with a null value, currently we ignore all null values

Reply:
Can't we mark Username and Password to send null all the time? 
I know it's autogenerated, but can we think about a workaround there? I would rather mess with the code generator than adding a boilerplate only for this.

Reply:
hmm, I'm not sure how. I could try adding a `JsonConverter` for this class

Reply:
once #477 is merged I'll add the JsonConverter for this

Reply:
#477 Merged.